core/admin/mailu/internal/views/autoconfig.py

```diff
@@ -0,0 +1,184 @@
+<username>%EMAILADDRESS%</username>
+<authentication>password-cleartext</authentication>
+</incomingServer>
+
+<outgoingServer type=\"smtp\">
+<hostname>{hostname}</hostname>
+<port>465</port>
+<socketType>SSL</socketType>
+<username>%EMAILADDRESS%</username>
+<authentication>password-cleartext</authentication>
+<addThisServer>true</addThisServer>
+<useGlobalPreferredServer>true</useGlobalPreferredServer>
+</outgoingServer>
+
+<documentation url=\"https://{hostname}/admin/client\">
+<descr lang=\"en\">Configure your email client</descr>
+</documentation>
+</emailProvider>
+</clientConfig>\r\n'''
+    return flask.Response(xml, mimetype='text/xml', status=200)
+
+@internal.route("/autoconfig/microsoft.json")
+def autoconfig_microsoft_json():
+    proto = flask.request.args.get('Protocol', 'Autodiscoverv1')
+    if proto == 'Autodiscoverv1':
+        hostname = app.config['HOSTNAME']
+        json = f'"Protocol":"Autodiscoverv1","Url":"https://{hostname}/autodiscover/autodiscover.xml"'
+        return flask.Response('{'+json+'}', mimetype='application/json', status=200)
+    else:
+        return flask.abort(404)
+
+@internal.route("/autoconfig/microsoft", methods=['POST'])
+def autoconfig_microsoft():
+    # https://docs.microsoft.com/en-us/previous-versions/office/office-2010/cc511507(v=office.14)?redirectedfrom=MSDN#Anchor_3
+    hostname = app.config['HOSTNAME']
+    try:
+        xmlRequest = (flask.request.data).decode("utf-8")
+        xml = xmltodict.parse(xmlRequest[xmlRequest.find('<'):xmlRequest.rfind('>')+1])
+        schema = xml['Autodiscover']['Request']['AcceptableResponseSchema']
+        if schema != 'http://schemas.microsoft.com/exchange/autodiscover/outlook/responseschema/2006a':
+            return flask.abort(404)
+        email = xml['Autodiscover']['Request']['EMailAddress']
+        xml = f'''<?xml version=\"1.0\" encoding=\"utf-8\" ?>
+    <Autodiscover xmlns=\"http://schemas.microsoft.com/exchange/autodiscover/responseschema/2006\">
+        <Response xmlns=\"{schema}\">
+            <Account>
+            <AccountType>email</AccountType>
+            <Action>settings</Action>
+            <Protocol>
+                <Type>IMAP</Type>
+                <Server>{hostname}</Server>
+                <Port>993</Port>
+                <LoginName>{email}</LoginName>
+                <DomainRequired>on</DomainRequired>
+                <SPA>off</SPA>
+                <SSL>on</SSL>
+            </Protocol>
+            <Protocol>
+                <Type>SMTP</Type>
+                <Server>{hostname}</Server>
+                <Port>465</Port>
+                <LoginName>{email}</LoginName>
+                <DomainRequired>on</DomainRequired>
+                <SPA>off</SPA>
+                <SSL>on</SSL>
+                </Protocol>
+            </Account>
+        </Response>
+    </Autodiscover>'''
+        return flask.Response(xml, mimetype='text/xml', status=200)
+    except:
+        pass
+    return flask.abort(400)
```

Comment:
You could move the return in the except clause.

Reply:
That's 83140322